fdbclient/DatabaseConfiguration.cpp

```diff
@@ -48,7 +48,7 @@ void parse( int* i, ValueRef const& v ) {
 	*i = atoi(v.toString().c_str());
 }
 
-void parseReplicationPolicy(IRepPolicyRef* policy, ValueRef const& v) {
+void parseReplicationPolicy(Reference<IReplicationPolicy>* policy, ValueRef const& v) {
```

Comment:
Why passing around a pointer of a ... reference?

Reply:
good catch!

Reply:
I looked a it a little more, and I believe this is so that if other people have a copy of the Reference<IReplicationPolicy>, they keep the original. This then creates a new reference counted object which policy points to.

Reply:
Ok now I see that `serializeReplicationPolicy` can actually deserialize binaries from a ... reader... The naming is kinda confusing  😂